slitlessutils/core/modules/extract/single/single.py

```diff
@@ -417,94 +496,226 @@ def extract(self, data, sources, **kwargs):
+                                    yy = yg[g]
+                                    ww = wav[g]
+                                    vv = val[g]
+                                    dw = dwav[g]
+
+                                    # apply the bitmask
+                                    xx, yy, vv, ww, dw = self.apply_bitmask(
+                                        dqa[yy, xx], xx, yy, vv, ww, dw, bitmask=bitmask)
+
+                                    # compute cross dispersion direction
+                                    # t = detdata.config[ordname].displ.invert(source.xyc[0],
+                                    # source.xyc[1], thiswave)
+                                    # dxdt = detdata.config[ordname].dispx.deriv(source.xyc[0],
+                                    # source.xyc[1], t)
+                                    # dydt = detdata.config[ordname].dispy.deriv(source.xyc[0],
+                                    # source.xyc[1], t)
+                                    # drdt = np.sqrt(dxdt**2 + dydt**2)
+
+                                    # dispx = dxdt/drdt
+                                    # dispy = dydt/drdt
+
+                                    # cross dispersion vector (orthogonal, by construction)
+                                    # crossx = -dy
+                                    # crossy = +dx
+
+                                    # center of region
+                                    # x0 = np.average(xx, weights=vv)
+                                    # y0 = np.average(yy, weights=vv)
+
+                                    # distances to the cross dispersion vector
+                                    # D = np.abs(crossx*(yy-y0)-crossy*(xx-x0))
+                                    # g = np.where(D<2)[0]
+                                    # if g.size>0:
+                                    #     xx = xx[g]
+                                    #     yy = yy[g]
+                                    #     vv = vv[g]
+                                    #     ww = ww[g]
+                                    #     dw = dw[g]
+
+                                    # measure size
+                                    # X = np.average(xx, weights=vv)
+                                    # Y = np.average(yy, weights=vv)
+                                    # X2 = np.average((xx-X)**2, weights=vv)
+                                    # Y2 = np.average((yy-Y)**2, weights=vv)
+                                    # XY = np.average((xx-X)*(yy-Y), weights=vv)
+                                    # DIF = (X2-Y2)/2.
+                                    # AVE = (X2+Y2)/2.
+                                    # DET = DIF**2 + XY**2
+                                    # A = np.sqrt(AVE + np.sqrt(DET))
+                                    # B = np.sqrt(AVE - np.sqrt(DET))
+                                    # SIG = np.sqrt(A*B)
+                                    # FWHM = SIG*2.35
```

Comment:
Yeah, it can.  But I think long-term we'll need something like this, and so I left it in commented (mostly so we won't have to look this stuff up again).  I'm fine either way, whatever you think is best.

Reply:
Gotcha, that's fine then. As long as it's there for a reason.